table/table_test.cc

```diff
@@ -2547,6 +2548,146 @@ void TableTest::IndexTest(BlockBasedTableOptions table_options) {
   c.ResetTableReader();
 }
 
+class BlockBasedTableFilterConstructCorruptionTest
```

Comment:
TODO: If it's okay to separate re-trying compaction into another PR, then in this PR, it's worth verifying that adding a new type of error in table builder does not break existing high-level write API Compact, Put, Delete

Reply:
Yes - it's okay to separate retrying compaction.